community/kernel/src/main/java/org/neo4j/helpers/TransactionTemplate.java

```diff
@@ -100,29 +125,47 @@ public TransactionTemplate( GraphDatabaseService gds, Monitor monitor, int retri
 
     public TransactionTemplate with( GraphDatabaseService gds )
     {
+        Objects.requireNonNull( gds );
         return new TransactionTemplate( gds, monitor, retries, backoff, retryPredicate );
     }
 
     public TransactionTemplate retries( int retries )
     {
+        if ( retries < 1 )
+        {
+            throw new IllegalArgumentException( "Number of retries must be greater than or equal to 1" );
+        }
```

Comment:
Oh, so the number of retries actually includes the first try as well.

Reply:
Passing 0 will result in an instant failure in `execute` due to the for loop. I should probably patch that now that you mention it.